src/react-apps/ux-editor/src/sagas/ruleConnection/ruleConnectionSagas.ts

```diff
@@ -76,28 +80,36 @@ function* checkIfRuleShouldRunSaga({ lastUpdatedDataBinding, lastUpdatedDataValu
       let numberOfInputFieldsFilledIn = 0;
       for (const inputParam in connectionDef.inputParams) {
         if (!inputParam) continue;
-        if (formFillerState.formData[connectionDef.inputParams[inputParam]]) {
+        let inputParamBinding: string = connectionDef.inputParams[inputParam];
+        if (isPartOfRepeatingGroup) {
+          inputParamBinding = inputParamBinding.replace(dataModelGroup, dataModelGroupWithIndex);
+        }
+        if (formFillerState.formData[inputParamBinding]) {
           numberOfInputFieldsFilledIn++;
         }
         if (connectionDef.inputParams[inputParam] === lastUpdatedDataBinding.DataBindingName) {
           shouldRunFunction = true;
         }
       }
       for (const outParam of Object.keys(connectionDef.outParams)) {
-        if (!outParam && !formFillerState.formData[connectionDef.outParams[outParam]]) {
```

Comment:
Isn't this an important check? Rule should not run if the form data is not there...

Reply:
Is the formData supposed to contain any data for a field that is not yet set? It does not appear to be the case as of now.

Reply:
Good point 👍